src/slurmdbd_ops.py

```diff
@@ -0,0 +1,447 @@
+        else:
+            logger.error(f"'{self._package_name}' not found on system.")
+
+        # Disable the ubuntu-hpc repo.
+        repositories = apt.RepositoryMapping()
+        repositories.disable(self._repo())
+
+        # Remove the key.
+        if self._keyring_path.exists():
+            self._keyring_path.unlink()
+
+    def upgrade_to_latest(self) -> None:
+        """Upgrade package to latest."""
+        try:
+            slurm_package = apt.DebianPackage.from_system(self._package_name)
+            slurm_package.ensure(apt.PackageState.Latest)
+            logger.info(f"Updated '{self._package_name}' to: {slurm_package.version.number}.")
+        except apt.PackageNotFoundError:
+            logger.error(f"'{self._package_name}' not found in package cache or on system.")
+        except apt.PackageError as e:
+            logger.error(f"Could not install '{self._package_name}'. Reason: {e.message}")
+
+    def version(self) -> str:
+        """Return the package version."""
+        try:
+            slurmdbd_vers = apt.DebianPackage.from_installed_package(self._package_name)
+        except apt.PackageNotFoundError:
+            logger.error(f"'{self._package_name}' not found on system.")
+        return slurmdbd_vers.version.number
+
+
+class SlurmdbdOpsManager:
+    """SlurmdbdOpsManager."""
+
+    def __init__(self):
+        """Set the initial attribute values."""
+        self._slurm_component = "slurmdbd"
+
+        self._slurm_state_dir = Path("/var/spool/slurmdbd")
+        self._jwt_rsa_key_file = self._slurm_state_dir / "jwt_hs256.key"
+        self._munge_key_path = Path("/etc/munge/munge.key")
+        self._slurmdbd_conf_dir = Path("/etc/slurm")
+        self._slurmdbd_conf_file = self._slurmdbd_conf_dir / "slurmdbd.conf"
+        self._slurmdbd_log_dir = Path("/var/log/slurm")
+        self._slurmdbd_log_file = self._slurmdbd_log_dir / "slurmdbd.log"
+        self._slurm_user = "slurm"
+        self._slurm_group = "slurm"
+        self._user_group = f"{self._slurm_user}:{self._slurm_group}"
+
+        self._munge_package = CharmedHPCPackageLifecycleManager("munge")
+        self._slurmdbd_package = CharmedHPCPackageLifecycleManager("slurmdbd")
+
+    def write_slurmdbd_conf(self, slurmdbd_parameters: dict) -> None:
+        """Render slurmdbd.conf."""
+        slurmdbd_conf = self._slurmdbd_conf_file
+
+        if slurmdbd_conf.exists():
+            slurmdbd_conf.write_bytes(os.urandom(2048))
+            slurmdbd_conf.unlink()
+
+        header = textwrap.dedent(
+            f"""
+            #
+            # {self._slurmdbd_conf_file} generated at {datetime.now()}
+            #
+
+            """
+        )
+        slurmdbd_conf.write_text(
+            header + "\n".join([f"{k}={v}" for k, v in slurmdbd_parameters.items() if v != ""])
+        )
+
+        slurmdbd_conf.chmod(0o600)
+
+        try:
+            subprocess.call(["chown", self._user_group, f"{slurmdbd_conf}"])
```

Comment:
Any particular reason why you're not using [`os.chown(...)`](https://docs.python.org/3/library/os.html#os.chown) here?

You can use the [`pwd`](https://docs.python.org/3/library/pwd.html) and [`grp`](https://docs.python.org/3/library/grp.html) modules to retrieve the uid and gid of the slurm user.